bots/mainbot/pilgrim.js

```diff
@@ -23,23 +24,41 @@ const dirs = [
     { x: 1, y: 1 },
     { x: -1, y: 1 },
 ];
+
+pilgrim.create_resource_map = (self) => {
+	let resource_map = new Array(self.map.length);
+	for (let i = 0; i < self.map.length; i++) {
+		resource_map[i] = new Array(self.map[0].length);
+	}
+
+	let i, j;
+  for (i = 0; i < self.map.length; i++) {
+		for (j = 0; j < self.map[i].length; j++) {
+			resource_map[i][j] = self.karbonite_map[i][j] || self.fuel_map[i][j];
+		}
+	}
+
+	pilgrim.resource_map = resource_map;
+}
 	
 // Find the closest resource deposit that has not been blacklisted to my 
 // current location
 // Parameters: 
-//		map - Either this.fuel_map or this.karbonite_map
 //		blacklist - A list of locations to ignore when searching for resources
 // Returns: Object with x and y properties specifying the coordinates of the
 // closest resource deposit
-pilgrim.findClosestResource = (self, map, blacklist=[]) => {
-  var closestLocation = undefined;
-	var closestDistance = Infinity;
-	
-	var i, j;
+pilgrim.findClosestResource = (self, blacklist=[]) => {
+  let closestLocation = undefined;
+	let closestDistance = Infinity;
 
-  for (i = 0; i < map.length; i++) {
-		for (j = 0; j < map[i].length; j++) {
-			if (map[i][j]) {
+	if (pilgrim.resource_map === undefined) {
```

Comment:
I like the idea of combining the maps, As long as there are some fuel and some karbonite around it should get mined.